library/multichoicerecyclerview/src/main/java/com/davidecirillo/multichoicerecyclerview/MultiChoiceAdapter.java

```diff
@@ -80,8 +83,14 @@ public void selectAll() {
      * @return true if has been selected false otherwise
      */
     public boolean select(int position) {
-        if (mIsInMultiChoiceMode || mIsInSingleClickMode && mItemList.containsKey(position)) {
-            perform(Action.SELECT, position, true, true);
+        if (mItemList.containsKey(position)) {
+            if (mIsInSingleClickMode || mIsInMultiChoiceMode) {
+                perform(Action.SELECT, position, true, true);
+                return true;
+            }
+        } else {
+            mItemList.put(position, State.INACTIVE);
```

Comment:
> because it can happen that the user add the list of item to the adapter later and at the onAttachToRecyclerView the getItemCount is 0

That's true. It's not perfect using `onAttachToRecyclerView()` to populate the internal list. We cannot use `notifyDatasetChanged()` neither as it's final.

So the only way to have the list updated seems to be either having a method the user needs to call or adding the positions like you do now...

---

I did not check the process (which method calls which) because I'm on the go currently but it looks fine and should be preferred to have this "workaround" only on one place. So +1 for `processUpdate`.